src/AavePolygonTreasuryMigration_20230801/AavePolygonTreasuryMigration_20230801.t.sol

```diff
@@ -0,0 +1,72 @@
+import {AaveV3Polygon, AaveV3PolygonAssets} from 'aave-address-book/AaveV3Polygon.sol';
+import {IERC20} from 'solidity-utils/contracts/oz-common/interfaces/IERC20.sol';
+import {AavePolygonTreasuryMigration_20230801} from './AavePolygonTreasuryMigration_20230801.sol';
+
+contract AavePolygonTreasuryMigration_20230801_Test is ProtocolV2TestBase {
+  function setUp() public {
+    vm.createSelectFork(vm.rpcUrl('polygon'), 45864548);
+  }
+
+  function testpayload() public {
+    address[] memory OLD_ASSETS_MIGRATED = new address[](10);
+    OLD_ASSETS_MIGRATED[0] = AaveV2PolygonAssets.DAI_A_TOKEN;
+    OLD_ASSETS_MIGRATED[1] = AaveV2PolygonAssets.USDC_A_TOKEN;
+    OLD_ASSETS_MIGRATED[2] = AaveV2PolygonAssets.USDT_A_TOKEN;
+    OLD_ASSETS_MIGRATED[3] = AaveV2PolygonAssets.WBTC_A_TOKEN;
+    OLD_ASSETS_MIGRATED[4] = AaveV2PolygonAssets.WETH_A_TOKEN;
+    OLD_ASSETS_MIGRATED[5] = AaveV2PolygonAssets.WMATIC_A_TOKEN;
+    OLD_ASSETS_MIGRATED[6] = AaveV2PolygonAssets.BAL_A_TOKEN;
+    OLD_ASSETS_MIGRATED[7] = AaveV2PolygonAssets.CRV_A_TOKEN;
+    OLD_ASSETS_MIGRATED[8] = AaveV2PolygonAssets.GHST_A_TOKEN;
+    OLD_ASSETS_MIGRATED[9] = AaveV2PolygonAssets.LINK_A_TOKEN;
+
+    address[] memory NEW_ASSETS_MIGRATED = new address[](10);
+    NEW_ASSETS_MIGRATED[0] = AaveV3PolygonAssets.DAI_A_TOKEN;
+    NEW_ASSETS_MIGRATED[1] = AaveV3PolygonAssets.USDC_A_TOKEN;
+    NEW_ASSETS_MIGRATED[2] = AaveV3PolygonAssets.USDT_A_TOKEN;
+    NEW_ASSETS_MIGRATED[3] = AaveV3PolygonAssets.WBTC_A_TOKEN;
+    NEW_ASSETS_MIGRATED[4] = AaveV3PolygonAssets.WETH_A_TOKEN;
+    NEW_ASSETS_MIGRATED[5] = AaveV3PolygonAssets.WMATIC_A_TOKEN;
+    NEW_ASSETS_MIGRATED[6] = AaveV3PolygonAssets.BAL_A_TOKEN;
+    NEW_ASSETS_MIGRATED[7] = AaveV3PolygonAssets.CRV_A_TOKEN;
+    NEW_ASSETS_MIGRATED[8] = AaveV3PolygonAssets.GHST_A_TOKEN;
+    NEW_ASSETS_MIGRATED[9] = AaveV3PolygonAssets.LINK_A_TOKEN;
+
+    uint256[] memory NEW_ASSETS_BEFORE_BALANCE = new uint256[](10);
+    uint256[] memory OLD_ASSETS_BEFORE_BALANCE = new uint256[](10);
+
+    uint256 i = 0;
+    uint256 payloadBalanceAfter;
+    uint256 collectorBalanceAfter;
+
+    for(; i < NEW_ASSETS_MIGRATED.length; i++) {
+      NEW_ASSETS_BEFORE_BALANCE[i] = IERC20(NEW_ASSETS_MIGRATED[i]).balanceOf(address(AaveV3Polygon.COLLECTOR));
+      OLD_ASSETS_BEFORE_BALANCE[i] = IERC20(OLD_ASSETS_MIGRATED[i]).balanceOf(address(AaveV2Polygon.COLLECTOR));
+    }
+
+    AavePolygonTreasuryMigration_20230801 payload = new AavePolygonTreasuryMigration_20230801();
+
+    GovHelpers.executePayload(vm, address(payload), AaveGovernanceV2.POLYGON_BRIDGE_EXECUTOR);
+
+    for(i = 0; i < OLD_ASSETS_MIGRATED.length; i++) {
+      payloadBalanceAfter = IERC20(OLD_ASSETS_MIGRATED[i]).balanceOf(address(payload));
+      assertEq(payloadBalanceAfter, 0);
+    }
+
+    for(i = 0; i < NEW_ASSETS_MIGRATED.length; i++) {
+      payloadBalanceAfter = IERC20(NEW_ASSETS_MIGRATED[i]).balanceOf(address(payload));
+      assertEq(payloadBalanceAfter, 0);
```

Comment:
@defijesus as the payload is called via a delegatecall, the tokens are actually transferred to the polygon bridge executor and not the payload, and then after the migration they are transferred back to the collector.

So we can test the balance of polygon bridge executor to be 0 instead of the payload as no tokens actually are transferred to the payload.

Reply:
can't believe I missed this completely, will make old and new asset comparisons on the bridge executor instead of the payload address. as the assets never actually touch the payload